src/plots/gl2d/scene2d.js

```diff
@@ -426,6 +427,13 @@ proto.plot = function(fullData, calcData, fullLayout) {
         ax.setScale();
     }
 
+    var mockLayout = {
+        _axisConstraintGroups: this.graphDiv._fullLayout._axisConstraintGroups,
+        xaxis: this.xaxis,
+        yaxis: this.yaxis
+    };
+    enforceAxisConstraints({_fullLayout: mockLayout});
```

Comment:
kind of hacky, but `enforceAxisConstraints ` has to happen after the `Axes.doAutoRange` just above, and these aren't the `gd._fullLayout` axes...

A bunch of things here would break if/when we support overlaid axes...